internal/gps/filesystem.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gps
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// filesystemState represents the state of a file system.
+type filesystemState struct {
+	root  string
+	dirs  []string
+	files []string
+	links []fsLink
+}
+
+// fsLink represents a symbolic link.
+type fsLink struct {
+	path string
+	to   string
+}
+
+// calculateFilesystemState returns a filesystemState based on the state of
+// the filesystem on root.
+func calculateFilesystemState(root string) (filesystemState, error) {
```

Comment:
I've been using `calculate` a lot in this context but I'm not very comfortable with it. Any better suggestions?

Reply:
`derive`, `analyze`, or even just `new`?